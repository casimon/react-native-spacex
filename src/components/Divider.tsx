import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle
} from 'react-native';

import colors from '../constants/colors';

const styles = StyleSheet.create({
  divider: {
    backgroundColor: colors.grey,
    width: 120,
    height: 4,
    alignSelf: 'center',
    marginVertical: 16,
    borderRadius: 2
  }
});

interface Props {
  style?: StyleProp<ViewStyle>;
}

class Divider extends React.PureComponent<Props> {
  render() {
    return <View style={[styles.divider, this.props.style]}></View>;
  }
}

export default Divider;
