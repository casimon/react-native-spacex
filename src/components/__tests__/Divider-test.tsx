/**
 * @format
 */

import 'react-native';
import React from 'react';
import Divider from '../Divider';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders divider correctly', () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders divider correctly with optional style prop', () => {
  const tree = renderer.create(<Divider style={{height: 4}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
