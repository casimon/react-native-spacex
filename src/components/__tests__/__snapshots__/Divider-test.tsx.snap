// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders divider correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "height": 4,
        "marginVertical": 16,
        "width": 120,
      },
      undefined,
    ]
  }
/>
`;

exports[`renders divider correctly with optional style prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "height": 4,
        "marginVertical": 16,
        "width": 120,
      },
      Object {
        "height": 4,
      },
    ]
  }
/>
`;
